src/wp-includes/class-wp-theme.php

```diff
@@ -374,13 +374,26 @@ public function __construct( $theme_dir, $theme_root, $_child = null ) {
 		}
 
 		// If we got our data from cache, we can assume that 'template' is pointing to the right place.
-		if ( ! is_array( $cache ) && $this->template != $this->stylesheet && ! file_exists( $this->theme_root . '/' . $this->template . '/index.php' ) ) {
+		if (
+			! is_array( $cache ) &&
+			$this->template != $this->stylesheet &&
+			! file_exists( $this->theme_root . '/' . $this->template . '/index.php' ) &&
+			! file_exists( $this->theme_root . '/' . $this->template . '/templates/index.html' ) &&
```

Comment:
This check looks similar to the check in [is_block_theme](https://github.com/WordPress/wordpress-develop/blob/acbbee8a1199f9ce70f1d3bbcc9cb80e32047e6c/src/wp-includes/class-wp-theme.php#L1493).  Is this by design?

Reply:
Per the linked ticket, I think this was picked up from similar code, so that would make sense that it's similar.

Looking at https://core.trac.wordpress.org/changeset/53416 it seems like it's using similar code over there too.